test_runner/regress/test_sharding.py

```diff
@@ -265,26 +266,13 @@ def shards_on_disk(shard_ids):
         pageserver.http_client().timeline_gc(tenant_shard_id, timeline_id, None)
     workload.validate()
 
-    migrate_to_pageserver_ids = list(
-        set(p.id for p in env.pageservers) - set(pre_split_pageserver_ids)
-    )
-    assert len(migrate_to_pageserver_ids) == split_shard_count - shard_count
-
-    # Migrate shards away from the node where the split happened
-    for ps_id in pre_split_pageserver_ids:
-        shards_here = [
-            tenant_shard_id
-            for (tenant_shard_id, pageserver) in all_shards
-            if pageserver.id == ps_id
-        ]
-        assert len(shards_here) == 2
-        migrate_shard = shards_here[0]
-        destination = migrate_to_pageserver_ids.pop()
+    # Enough background reconciliations should result in the shards being properly distributed
+    env.storage_controller.reconcile_until_idle()
 
-        log.info(f"Migrating shard {migrate_shard} from {ps_id} to {destination}")
-        env.storage_controller.tenant_shard_migrate(migrate_shard, destination)
-
-    workload.validate()
+    # We have 8 shards and 16 nodes
+    # Initially I expect 4 nodes to have 2 attached locations each, and another 8 nodes to have
+    # 1 secondary location each
+    # 2 2 2 2 1 1 1 1 1 1 1 1 0 0 0 0
```

Comment:
Would be nice to spell out the expected state and validate for it (instead of just counting reconciles)

Reply:
Made this stricter in 1cb55ed460f1401b453c27069687c82f031807f5

The result is pretty simple (just one location on every node), but I've spelled it out as a dict so that it's obvious which node is at fault if it fails.